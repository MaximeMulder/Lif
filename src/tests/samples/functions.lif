function twenty(number: Integer): Integer {
    return number + 20;
}

print(twenty(22));

function greet(greeting: String, ...persons: Array[String]) {
	for person in persons {
		print(greeting + " " + person + " !");
	}
}

greet("Hello", "Alice", "Bob");

function callback(variable) {
    print("Variable " + variable + " from callback.");
}

function executor(callback) {
    callback(42);
}

executor(callback);
