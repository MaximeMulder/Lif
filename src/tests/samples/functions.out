42
Hello Alice !
Hello Bob !
Variable 42 from callback.
